### ISSUE TITLE
Waydroid rendering is borked

### ISSUE THREAD

**Issue by TheSunCat** [hyprwm/Hyprland#2190]
Hi! I tried setting up Waydroid and it seems to have really bad rendering issues, especially when fullscreened. It seems that only a portion of the window gets damage events, while the right and bottom sides get updated late. Additionally, the mouse cursor does not match the cursor on Hyprland. These issues do not appear on Sway.

Hyprland:

https://user-images.githubusercontent.com/44881120/235318315-8266d510-268e-49de-b371-745aa907edfc.mp4

Sway:

https://user-images.githubusercontent.com/44881120/235318222-a74510f5-462f-4808-a50f-5b59ccbdad69.mp4

**Comment by TheComputerM** [comment]
Same issue with gtk dialogs on some apps like virt-manager

**Comment by dragove** [comment]
I've got the same issue. the default size of waydroid (fullscreen) works fine. But I have two monitors with different resolution(1080P scale 1 and 4k scale 1.5) then waydroid gets broken if I move it from one monitor to another.

**Comment by univeous** [comment]
The mouse cursor offset problem also appears in some games through Wine/Proton in fullscreen.

**Comment by TheSunCat** [comment]
> The mouse cursor offset problem also appears in some games through Wine/Proton in fullscreen.

Do you know which games? I can try to repro on my end (this is still happening and prevents me from using Waydroid entirely). Also, are you using `force_zero_scaling`? 

> Same issue with gtk dialogs on some apps like virt-manager

Sounds like #2704

**Comment by univeous** [comment]
Enable `force_zero_scaling` does not help.

This problem mainly occurs in some VNs, such as [CHAOS; CHILD](https://store.steampowered.com/app/970570/CHAOSCHILD/).

Setting it to full screen in the game and starting it will not work, it has some strange stretch:
![image](https://github.com/hyprwm/Hyprland/assets/40604180/0e14b3b4-68b2-4c86-808a-a783f2b34c06)

If starting the game in windowed mode and manually full screen, it looks good but there is the cursor offset issue.

I'm using `hyprland-nvidia-git`, not sure if it matters.

**Comment by izmyname** [comment]
Is this still borked?

**Comment by TallenPeli** [comment]
> Is this still borked?

Yes, seems that Hyprland and waydroid don't work very well together.
(on intel graphics)

![Pasted image](https://github.com/user-attachments/assets/f5104764-751b-417a-a42e-4b315ee68196)

**Comment by MillionsToOne** [comment]
You have to manually resize the window prop for it to work. Like this: ```
waydroid prop set persist.waydroid.width "1920" 
waydroid prop set persist.waydroid.height "1080"
```

**Comment by matejdro** [comment]
That only works with a single WayDroid window that spans the entire screen. It does not work with individual app-per-window

**Comment by yvvki** [comment]
Another issue with HiDPI scaling. Setting the `persist.waydroid.*` to my native resolution with my monitor scaling set to number larger than 1 leads me to the display being much bigger than my screen.

![2024-11-26-002358_hyprshot](https://github.com/user-attachments/assets/49101ce4-5de6-4993-a4d4-e2a245977f1f)  
(Notice the menu being cut off.)

**Comment by ZakariaBouzouf** [comment]
I still have this problem , it seems that the `persist.waydroid.*` params don't change when i check with the `sudo waydroid shell getprop` command. Any solution ?

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
